packages/react-native-analytics/src/integrations/castle/__tests__/Castle.test.js

```diff
@@ -0,0 +1,274 @@
+import CastleReactNativeIntegration, {
+  LOGGER_MESSAGE_PREFIX,
+  HTTP_CLIENT_ERROR_MESSAGE,
+} from '../';
+import Integration from '../../integration';
+import CastleIO from '@castleio/react-native-castle';
+import coreClient from '@farfetch/blackout-core/helpers/client';
+import utils from '@farfetch/blackout-core/analytics/utils';
+
+const mockRequestHeaderName = 'castle-header-foo';
+const mockRequestHeaderValue = '12342342345241342423424';
+jest.mock('@castleio/react-native-castle', () => {
+  return {
+    ...jest.requireActual('@castleio/react-native-castle'),
+    configure: jest.fn(() => Promise.resolve()),
+    requestTokenHeaderName: async () => Promise.resolve(mockRequestHeaderName),
+    createRequestToken: async () => Promise.resolve(mockRequestHeaderValue),
+    identify: jest.fn(() => Promise.resolve()),
+    reset: jest.fn(() => Promise.resolve()),
+  };
+});
+
+jest.mock('@farfetch/blackout-core/analytics/utils', () => ({
+  ...jest.requireActual('@farfetch/blackout-core/analytics/utils'),
+  logger: {
+    error: jest.fn(),
+    warn: jest.fn(),
+  },
+}));
+
+const mockIntegrationOptions = {
+  httpClient: {
+    defaults: {
+      headers: {
+        common: {},
+      },
+    },
+  },
+  configureHttpClientHeader: jest.fn(),
+};
+const getIntegrationInstance = async customOptions => {
+  return await CastleReactNativeIntegration.createInstance(
+    customOptions || mockIntegrationOptions,
+  );
+};
+
+describe('Castle', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Integration', () => {
+    it('should extend the default Integration class', async () => {
+      const instance = await getIntegrationInstance();
+
+      expect(instance).toBeInstanceOf(CastleReactNativeIntegration);
+      expect(instance).toBeInstanceOf(Integration);
+    });
+
+    it('should not depend on the consent to be loaded (required)', () => {
+      expect(CastleReactNativeIntegration.shouldLoad()).toBe(true);
+      expect(CastleReactNativeIntegration.shouldLoad(null)).toBe(true);
+      expect(
+        CastleReactNativeIntegration.shouldLoad({
+          statistics: false,
+          marketing: false,
+          preferences: false,
+        }),
+      ).toBe(true);
+    });
+  });
+
+  describe('Instance', () => {
+    it('should have a property referring the Castle.io instance', async () => {
+      const instance = await getIntegrationInstance();
+
+      expect(instance.castleIO).toBe(CastleIO);
+    });
+
+    describe('HTTP client', () => {
+      it('should have the default one assigned if none passed', async () => {
+        const instance = await CastleReactNativeIntegration.createInstance({});
```

Comment:
This could use the `getIntegrationInstance` method the same way as discussed here: https://github.com/Farfetch/blackout-react-native/pull/31/files#r812743640

Reply:
fixed 🙂  thanks!